### PR TITLE
K8s x APM

### PR DESCRIPTION
**Pull request by cswatt**
### What does this PR do?
Revamped instructions for setting up APM on K8s for both Daemonset and Helm installations.

### Motivation
engineering request

### Preview link

https://docs-staging.datadoghq.com/cswatt/k8s-with-apm/agent/kubernetes/daemonset_setup/#apm-and-distributed-tracing
https://docs-staging.datadoghq.com/cswatt/k8s-with-apm/agent/kubernetes/helm/?tab=macoshomebrew#enable-apm-and-distributed-tracing

### CODE REVIEW

File: content/en/agent/kubernetes/daemonset_setup.md [hunk 1]
```diff
@@ -201,15 +201,39 @@ To enable [Log collection][8] with your DaemonSet:
...
+        - containerPort: 8126
+          hostPort: 8126
+          name: traceport
+          protocol: TCP
+(...)
+```
+
+Use the downward API to pull the host IP; the application container needs an environment variable that points to `status.hostIP`. You can name it `DD_AGENT_HOST`:
+
+```
```
Comment (DylanLovesCoffee) [inline content/en/agent/kubernetes/daemonset_setup.md]: Indentation needed here so the markdown code block doesn't break.

File: content/en/agent/kubernetes/helm.md [hunk 2]
```diff
@@ -138,9 +138,51 @@ datadog:
...
+  apmEnabled: true
+
+(...)
+
+daemonset:
+  (...)
+  useHostPort: true
+```
+
+Use the downward API to pull the host IP; the application container needs an environment variable that points to `status.hostIP`. You can name it `DD_AGENT_HOST`. Update the `env` section of your `datadog-values.yaml` file with the following:
```
Comment (DylanLovesCoffee) [inline content/en/agent/kubernetes/helm.md]: For the helm chart configuration, this is actually being done on the user’s application side. So we might want to say something similar to what you did for the Daemonset section. Maybe something like `Update the env section of your application's manifest, or values.yaml file with the following:`

File: content/en/agent/kubernetes/helm.md [hunk 3]
```diff
@@ -138,9 +138,51 @@ datadog:
...
+env:
+  - name: DD_AGENT_HOST
+    valueFrom:
+      fieldRef:
+        fieldPath: status.hostIP
+```
+
+Then upgrade your Datadog Helm chart.
+
+Finally, point your application-level tracers to where the Datadog Agent host is using the environment variable `DD_AGENT_HOST`. For example, in Python:
```
Comment (DylanLovesCoffee) [inline content/en/agent/kubernetes/helm.md]: Similar to my previous comment [here](https://github.com/DataDog/documentation/pull/4352/files#r273689254). We can say: `Finally, point your application-level tracers to the host IP using the environment variable `DD_AGENT_HOST`

File: content/en/agent/kubernetes/daemonset_setup.md [hunk 4]
```diff
@@ -201,39 +201,70 @@ To enable [Log collection][8] with your DaemonSet:
...
     ```
 
-Learn more about this in [the Docker log collection documentation][9].
+Learn more about this in [the Docker log collection documentation][10].
 
-### Trace Collection
+### APM and Distributed Tracing
 
-To enable trace collection with your DaemonSet:
+To enable APM, set the `DD_APM_NON_LOCAL_TRAFFIC` variable to true in your *env* section:
```
Comment (apigirl) [inline content/en/agent/kubernetes/daemonset_setup.md]:
```suggestion
To enable APM by to allowing incoming data from port 8126, set the `DD_APM_NON_LOCAL_TRAFFIC` variable to true in your *env* section:
```
I would just add this up here and then delete the sentence below the example

File: content/en/agent/kubernetes/daemonset_setup.md [hunk 5]
```diff
@@ -201,39 +201,70 @@ To enable [Log collection][8] with your DaemonSet:
...
-          containers:
-          - name: <CONTAINER_NAME>
-            image: <CONTAINER_IMAGE>/<TAG>
-            env:
-              - name: DD_AGENT_HOST
-                valueFrom:
-                  fieldRef:
-                    fieldPath: status.hostIP
-    ```
+This is needed for the container to allow incoming data from port 8126.
```
Comment (apigirl) [inline content/en/agent/kubernetes/daemonset_setup.md]:
```suggestion
```

File: content/en/agent/kubernetes/daemonset_setup.md (hunk 1 above)
Comment (burningion) [reply content/en/agent/kubernetes/daemonset_setup.md]: Potential wording fix here:

Use Kubernetes' downward API to pull the host Node's IP with `status.hostIP`.  The Datadog container Agent expects this to be named `DD_AGENT_HOST`:

Note the `can name it`  isn't really appropriate. It must be named that to work.

File: content/en/agent/kubernetes/daemonset_setup.md [hunk 6]
```diff
@@ -201,39 +201,68 @@ To enable [Log collection][8] with your DaemonSet:
...
+import os
+from ddtrace import tracer
+
+tracer.configure(
+    hostname=os.environ['DD_AGENT_HOST'],
+    port=os.environ['DD_TRACE_AGENT_PORT'],
+)
+```
 
-    **Warning**: The `hostPort` parameter opens a port on your host. Make sure your firewall only allows access from your applications or trusted sources.
```
Comment (l0k0ms) [inline content/en/agent/kubernetes/daemonset_setup.md]: Should we remove the warning? if so, should we remove it from the DogStatsD section below as well?

**Comment by l0k0ms** [comment]
Just left a comment about the warning, i'm not so sure if we should keep it. Seems obvious to me but idk.

Regarding the helm APM documentation we should also display it here: https://github.com/helm/charts/tree/master/stable/datadog

**Pull request merged**